### PR TITLE
Use a single webpack instance

### PR DESCRIPTION
**Pull request by jbasdf**

### CODE REVIEW

File: client/config/webpack.config.js [hunk 1]
```diff
@@ -18,19 +20,20 @@ const _                    = require('lodash');
 //      'staging' (Will result in same output as production but will output to /staging)
 //      'hot' (Development mode with hot reload)
 //      'test'
-module.exports = function webpackConfig(app) {
+const outputSourceMaps = true;
```
Comment (dittonjs) [inline client/config/webpack.config.js]: was this supposed to be true always?

File: client/libs/build/apps.js [hunk 2]
```diff
@@ -1,27 +1,22 @@
...
 
 const settings = require('../../config/settings');
 const build = require('./build');
 const log = require('./log');
+const webpackConfigBuilder = require('../../config/webpack.config');
 
 // -----------------------------------------------------------------------------
 // Build a single app
 // -----------------------------------------------------------------------------
 function buildAppParts(app, onlyPack) {
```
Comment (dittonjs) [inline client/libs/build/apps.js]: remove this

File: client/webpack.hot.js [hunk 3]
```diff
@@ -63,26 +56,22 @@ if (appName) {
...
+  setupMiddleware(serverApp, results.webpackCompiler);
+  runServer(serverApp, settings.hotPort, settings.paths.devOutput);
 } else {
   // Run and serve all applications
-  clientApps.buildApps(options).then((results) => {
-    _.each(results, (result) => {
-      result.buildPromise.then(() => launch(result.app));
-    });
+  const results = clientApps.buildApps(options);
+  // results.webpackCompiler.run(() => {
```
Comment (dittonjs) [inline client/webpack.hot.js]: remove these comments

**Comment by bfcoder** [comment]
A few things to fix, but looks good to me.

File: client/config/webpack.config.js (hunk 1 above)
Comment (dittonjs) [reply client/config/webpack.config.js]: yes is it

File: client/libs/plugins/flow_compiler.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    });
+    compiler.plugin('invalid', () => {
+      this.promise = this.flowCheckTypes();
+    });
+  }
+
+  flowCheckTypes() {
+    return new Promise((resolve) => {
+      execFile(
+        require('flow-bin'),
```
[the remark this answers is not in the document]
Comment (dittonjs) [reply client/libs/plugins/flow_compiler.js]: need it

**Pull request merged**